src/app/CASESessionManager.h

```diff
@@ -63,6 +63,7 @@ class CASESessionManager : public Dnssd::OperationalResolveDelegate
     virtual ~CASESessionManager() { mDNSResolver.Shutdown(); }
 
     CHIP_ERROR Init();
+    void Shutdown() { mDNSResolver.Shutdown(); }
```

Comment:
Is calling `mDSNResolver.Shutdown()` twice (once here, once in the destructor) ok?

Reply:
Our shutdown methods generally accept multiple calls. Did not want to delete the destructor case to not break any existing code. Existing shutdown returns early if already shut down:

```
        VerifyOrReturn(mDelegate != nullptr);
        mDelegate->SetOperationalDelegate(nullptr);
        mDelegate->SetCommissioningDelegate(nullptr);
        mDelegate->Release();
        mDelegate = nullptr;
 ```

Reply:
Also note below that I could NOT reliably fix TestCommissioningManager but did not want to hold up all the other fixes on it. We have some sort of leak of udp endpoints in that test.